### PR TITLE
Update module version to v3

### PR DESCRIPTION
**Pull request by nprokopic**
### What does this PR do?

This PR bumps go module version to v3, due to a breaking change that has been added here https://github.com/giantswarm/kubectl-gs/pull/1352.

### What is the effect of this change to users?

See https://github.com/giantswarm/kubectl-gs/pull/1352.

### What does it look like?

See https://github.com/giantswarm/kubectl-gs/pull/1352.

### Any background context you can provide?

https://github.com/giantswarm/roadmap/issues/3466

### Do the docs need to be updated?

Will check.

### Should this change be mentioned in the release notes?

- [x] CHANGELOG.md has been updated

### Is this a breaking change?

(Breaking changes are, for example, removal of commnds/flags or substantial changes in the meaning of a flag. If yes, please add the `breaking-change` label to the PR.)

Yes.

### CODE REVIEW

**Comment by nprokopic** [comment]
cc @ssyno if you would like to merge this first https://github.com/giantswarm/kubectl-gs/pull/1212 and then we can rebase this PR.

**Review by njuettner — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ### Changed
 
-- CAPA only change for new releases: render release version in config instead of cluster-aws version in App resource.
+- **BREAKING** CAPA only change for new releases: render release version in config instead of cluster-aws version in App resource.
```
Comment (marians) [inline CHANGELOG.md:16]: I have to admit I don't really understand the changelog entry. Can we explain this in more natural language? Also, which commands are affected?

Comment (nprokopic) [reply CHANGELOG.md]: I have updated changelog so it's more clear.

**Review by marians — APPROVED** [review]
Thank you!

**Pull request merged**